src/platform/Linux/bluez/Helper.cpp

```diff
@@ -1735,25 +1737,48 @@ bool BluezUnsubscribeCharacteristic(BLE_CONNECTION_OBJECT apConn)
 
 struct ConnectParams
 {
-    ConnectParams(BluezDevice1 * device, BluezEndpoint * endpoint) : mDevice(device), mEndpoint(endpoint) {}
+    ConnectParams(BluezDevice1 * device, BluezEndpoint * endpoint) : mDevice(device), mEndpoint(endpoint), mNumRetries(0) {}
     BluezDevice1 * mDevice;
     BluezEndpoint * mEndpoint;
+    uint16_t mNumRetries;
 };
 
-static void ConnectDeviceDone(GObject * aObject, GAsyncResult * aResult, gpointer)
+static void ConnectDeviceDone(GObject * aObject, GAsyncResult * aResult, gpointer apParams)
 {
-    BluezDevice1 * device = BLUEZ_DEVICE1(aObject);
-    GError * error        = nullptr;
-    gboolean success      = bluez_device1_call_connect_finish(device, aResult, &error);
+    BluezDevice1 * device  = BLUEZ_DEVICE1(aObject);
+    GError * error         = nullptr;
```

Comment:
Minor suggestion: consider `std::unique_ptr<GError, GErrorDeleter> err` (and other `G*Deleter` w/ unique_ptr? And use a unique_ptr to hold the params so we can avoid the `exit:` & `ExitNow()` in this function.

Reply:
Perhaps spawn a new issue from this comment to track the request?

Reply:
Done in (https://github.com/project-chip/connectedhomeip/issues/18286)